src/block/Anvil.php

```diff
@@ -94,4 +101,26 @@ public function place(BlockTransaction $tx, Item $item, Block $blockReplace, Blo
 	public function tickFalling() : ?Block{
 		return null;
 	}
+
+	public function onHitGround(FallingBlock $blockEntity) : void{
+		if($this->position->isValid()){
+			foreach($this->position->getWorld()->getCollidingEntities($blockEntity->getBoundingBox()) as $ent){
+				if($ent instanceof Living){
+					if($blockEntity->fallDistance > 1){
+						//If player has helmet do 1/4 the normal damage
+						$helmet = $ent->getArmorInventory()->getHelmet();
+						if(!$helmet->isNull() && $helmet instanceof Durable){
+							$damageDone = $blockEntity->fallDistance * 0.5;
+							$ent->damageItem($helmet, (int) $damageDone);
```

Comment:
I don't like this. At this point the damage has not yet been done and might be cancelled. This denies plugins the opportunity to prevent the damage to health and armour via EntityDamageEvent cancelling.